asserts/ifacedecls.go

```diff
@@ -98,42 +111,94 @@ func compileMapAttrMatcher(cc compileContext, m map[string]interface{}) (attrMat
 	return matcher, nil
 }
 
-func matchEntry(apath, k string, matcher1 attrMatcher, v interface{}) error {
+func matchEntry(apath, k string, matcher1 attrMatcher, v interface{}, ctx AttrMatchContext) error {
 	apath = chain(apath, k)
-	if v == nil {
+	if _, ok := matcher1.(missingAttrMatcher); !ok && v == nil {
```

Comment:
Shouldn't that be done in the other (non-missing) constraint implementation themselves?